### ISSUE TITLE
Useless error message upon trying to save a doc as PDF (version 3.5.4 on Win 11 64 bit)

### ISSUE THREAD

**Issue by kpeters58** [RickStrahl/MarkdownMonster#1154]
What do I do with this rather useless error message??

![image](https://github.com/user-attachments/assets/23ab7832-dccb-4e9d-b145-faab7eb06615)

**Comment by RickStrahl** [comment]
The error message comes from the PDF engine and unfortunately it's not very specific. 

problem here is pretty obvious - 10" margins aren't going to work on a Letter sized document :smile:

Probably is a good idea to constrain those values to what the browser allows (don't think the actual limits are published).

**Comment by RickStrahl** [comment]
This is better, eh?

![image](https://github.com/user-attachments/assets/5ed590db-9b79-4efc-9fae-089451ddffa5)

**Comment by kpeters58** [comment]
Yes, that's how it's done properly. In this particular case, even a 'One or more margins too big' would have sufficed.

Thanks for the super-prompt reaction!!!
KP

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

**Comment by kpeters58** [comment]
Thanks for putting this in the latest release!

But something ain't quite right:

[image.png]

At first, I provoked the issue by setting 10 inches left & right and ran into the proper new error message. After that, I switched back to mm and tried to generate the PDF -
somehow the switchback to mm didn't properly register?

Kind regards,
KP

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

**Comment by RickStrahl** [comment]
I can't see the image you have there... 

> somehow the switchback to mm didn't properly register?

Not sure what that means. If you're saying you went back and cleared the numbers, I can't duplicate what you're saying can you give some more precise steps?

**Comment by kpeters58** [comment]
Hi Rick,

Here you go (still broken, tested today with 3.5.6.2):

[image.png]

Note that my assumption is that 1 cm all around should be easily accommodatable without triggering the new, better defined error message.

How to get there:

- In order to test your previous fix which improved upon the displayed error message, I triggered the message on purpose by selecting inches here with the 10 all around margins - this gave me the expected, new & improved message - so far, so good.
- I then did nothing else but change the margin units from Inches to Millimeters, leaving the values of 10 in all 4 edit boxes as they were and tried to generate the PDF again - it still failed with the same error message.
- From that i deduce, that the unit change from the combobox did not register with that part of your code which evaluates the margins to see if they fit, as 1 cm all around would fit easily.
- Furthermore, you seem to store the user's last used Margin unit selection (registry/inifile?) - so every time I now start up MDM and go to this screen, Millimeters is preselected as are the values of 10 in all edit boxes and PDF generation fails.

Hope that clears it up. If not, don't hesitate to hit me up for more...

Kind regards,
KP

Sent with [Proton Mail](https://proton.me/mail/home) secure email.
